sys/platform/pc64/x86_64/efirt.c

```diff
@@ -464,6 +501,130 @@ efi_get_table(struct uuid *uuid, void **ptr)
 	return (ENOENT);
 }
 
+int
+get_table_length(enum efi_table_type type, size_t *table_len, void **taddr)
+{
+	switch (type) {
+	case TYPE_ESRT:
+	{
+		struct efi_esrt_table *esrt = NULL;
+		struct uuid uuid = EFI_TABLE_ESRT;
+		uint32_t fw_resource_count = 0;
+		size_t len = sizeof(*esrt);
+		int error;
+		void *buf;
+
+		error = efi_get_table(&uuid, (void **)&esrt);
+		if (error != 0)
+			return (error);
+
+		buf = kmalloc(len, M_TEMP, M_WAITOK);
+		//Temporary solution - maybe copyout instead of physcopyout
+		//error = physcopyout((vm_paddr_t)esrt, buf, len);
```

Comment:
You might be able to copy `physcopyout` from https://github.com/freebsd/freebsd-src/blob/373ffc62c158e52cde86a5b934ab4a51307f9f2e/sys/kern/subr_uio.c#L119-L138

Otherwise need to temporary map the table into virtual address space: `bus_space_map` + `bus_space_vaddr` + `copyout` + `bus_space_unmap`. This can serve as an implementation for `physcopyout`.